src/main/java/ch/wisv/areafiftylan/web/sponsor/controller/SponsorController.java

```diff
@@ -78,4 +80,10 @@ public SponsorController(SponsorService sponsorService) {
         sponsorService.deleteAllSponsors();
         return createResponseEntity(HttpStatus.OK, "Successfully deleted all sponsors.");
     }
+
+    @ExceptionHandler(DataIntegrityViolationException.class)
+    public ResponseEntity<?> handleDataIntegrityViolationException(DataIntegrityViolationException ex) {
+        return createResponseEntity(HttpStatus.CONFLICT,
+                "Could not delete this sponsor because it is used by another entity!");
```

Comment:
This assumes only the DELETE call will trigger a `DataIntegrityViolation`. I get what you're trying to do, but this may lead to extemely confusing results in other cases. For example duplicate key entries will give the same exception but have nothing to do with deleting things. 

I recommend a response like "Could not complete request due to a constraint violation" or something. As the user calls this explicitly, it should make sense what the cause of the problem is.

Reply:
Hmm that's a good point.

I might be able to walk through the root cause of this exception, check if it contains `could not delete` (or something similar), and then send a `HttpStatus.CONFLICT` with this message, and otherwise a `HttpStatus.BAD_REQUEST` such that the error message is generic in the frontend. Sounds good?

Reply:
That could work yes, try not to make it too hacky 😉